src/celeritas/UnitTypes.hh

```diff
@@ -0,0 +1,265 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/UnitTypes.hh
+//! \brief Annotated unit struct definitions for use with Quantity
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "celeritas_config.h"
+
+#include "Constants.hh"
+#include "Units.hh"
+
+namespace celeritas
+{
+namespace units
+{
+//---------------------------------------------------------------------------//
+//!@{
+//! \name Natural units
+
+struct CLight
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+        return constants::c_light;
+    }
+    static char const* label() { return "c"; }
+};
+
+//! "Natural units" for energy
+struct Mev
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+#if CELERITAS_UNITS == CELERITAS_UNITS_CLHEP
+        return units::megaelectronvolt;
```

Comment:
Why the `#if` here and not in `Units.hh`?  I.e. why not define `units::megaelectronvolt` for the non-CLHEP cases?

Reply:
In all unit systems that use "macro scale" quantities (CGS and SI), `megaelectronvolt` requires a physical constant (e_electron) that's defined in `Constants.hh` (which is dependent on `Units.hh`). CLHEP is the weird exception because it indirectly defines charge and mass based on the value of `e_electron`.

Reply:
I am really confused :( ... the else part really seems to read Mega (1e6) Electron Volt, so I don't see why it is not in unit as `megaelectronvolt` ...

Reply:
```c++
#if CELERITAS_UNITS == CELERITAS_UNITS_CLHEP
        return units::megaelectronvolt;
#else
        return real_type(1e6) * constants::e_electron * units::volt;
#endif
```

CLHEP is the only system that defines MeV as a "unit" (value of 1). The other unit systems require a fundamental constant (which *can't* be included by Units.hh).

Reply:
> as a "unit" (value of 1)....Units.hh

Humm .. another ambiguous name it seems ie. it is `UnitValues.hh` but it could have been `UnitSystem.hh`